Added Slip Stalk to combat_trainer & common

intangibles opened the pull request:
You can now supply a YAML setting of "hide_type" to change what command is performed.  For example "hide_type: stalk" will perform slip stalk.

Also updated hide? method to use optional paramaters. Since hide? is in common, i wanted to retain its functionality for other scripts that may use it.

I tested the exp unofficially and it seemed to be better, but at the very least it has 25% reduced RT. Hide is 2 sec, stalk is 2 sec. Slip stalk is 3 sec total.

Yaml Settings:
hide_type: stalk     - will perform slip stalk instead of hide
force_unhide:  true  (default false) - will always unhide after hiding via Stealth timer.  (this will not affect attacks)

jandersson commented:
What does slip stalk do?

jandersson commented:
Nevermind, figured it out.
```
R>slip help                                                                                              
You begin to target a mountain giant.                                                                    
>                                                                                                        
1)  SLIP <right|left> or SLIP <item> in <container>       [Silent stowing of an item into your packs]    
2)  SLIP <item>                                           [Silent getting of an item from your packs]    
 - Addition 1:    Works on WORN items.                                                                   
 - Addition 2:    Works with items ON THE GROUND.                                                        
3)  STALK <person|creature> (while unhidden)              [Hide + Stalk of target]                       
4)  SNEAK <direction>    (while unhidden)                 [Hide + Sneak in direction]                    
```
```
>stalk                                                                            
You turn to face a mountain giant.                                                
You quickly slip into hiding to prepare to stalk.                                 
You melt into the background, convinced that your attempt to hide went unobserved.
You are convinced that your stalking went unobserved.                             
You move into position to stalk a mountain giant when it moves.                   
Roundtime: 3 sec.                                                                 
```

rcuhljr commented:
If the only point to hiding is training stealth I think you get more exp from hide+stalk then you do from using slip. If that's not the case it would be good to know.

rcuhljr commented on common.lic:
```diff
@@ -268,9 +268,10 @@ module DRC
+      case bput(hide_type, 'Roundtime', 'too busy performing', 'can\'t see any place to hide yourself', 'Stalk what')
       when 'too busy performing'
         bput('stop play', 'You stop playing', 'In the name of')
         return hide?
```
This needs to pass hide_type through again.

intangibles replied:
Ahh.  Great, I'm going to add another piece too for "You are already stalking."

```
when 'You're already stalking'
puts 'stop stalking'  #This shouldn't have any RT associated so waiting for a bput would slow the process down
return hide?(hide_type)
```

rcuhljr commented on common.lic:
```diff
@@ -268,9 +268,10 @@ module DRC
 
-  def hide?
+  def hide?(*hide_type)
+    hide_type.any? ? hide_type = hide_type[0] : hide_type = 'hide' 
```
We can replace both of these lines with 
```
def hide?(hide_type = 'hide')
```

and add hide_type to `base.yaml` and `base-empty.yaml` with the default value.

intangibles replied:
Great. i was trying to find a better way to do this but had no idea how!

intangibles replied:
Wasn't sure what to put here since these all seem to be lists

base-empty.yaml

  gear_sets: {}
  lichbot_buffs: []
  listen_skills: []
  offensive_spell_cycle: []
  ignored_npcs: []
  hunting_buddies: []
  empty_hunting_room_messages: []
  classes_to_teach: []
  hide_type:

and for base.yaml
hide_type:
or 
hide_type: hide

rcuhljr replied:
`hide_type: hide` for both of them. The goal is just to safe guard them if they decide to overwrite their `hide_type: stalk` with `hide_type:` in a later file.

rcuhljr replied:
Although this is drifting farther from the purpose of base-empty perhaps, maybe we should just let them error out and fix it. @rpherbig opinions?

rpherbig replied:
Sorry, I apparently missed this conversation. We resolved it below.

intangibles commented:
I tested the exp unofficially and it seemed to be better, but its also 25% reduced RT.  Hide is 2 sec, stalk is 2 sec.  Slip stalk is 3 sec total.

However, if your stealth timer is too short you will get a "You are already stalking message." I was thinking about adding "stop stalk" but if your stealth is around 15 seconds then stalk automatically drops.

> [combat-trainer]>stalk
> 
> You quickly slip into hiding to prepare to stalk.
> You melt into the background, convinced that your attempt to hide went unobserved.
> You are convinced that your stalking went unobserved.
> You move into position to stalk a warklin mauler when it moves.
> Roundtime: 3 sec.
> 
> HR> 
> You are revealed from concealment.
> The greenish radiation around you burbles warningly before a sphere of hissing acid spews forth in all directions!
> The wave lightly pits a warklin mauler's abdomen as it blasts across it.
> The wave lightly pits an armored warklin's abdomen as it blasts across it.
> The wave lightly pits an armored warklin's neck as it blasts across it.
> The wave lightly pits an armored warklin's chest as it blasts across it.
> The wave lightly pits an armored warklin's chest as it blasts across it.
> 
> R> 
> An armored warklin carefully picks its way west.
> 
> > 
> [combat-trainer]>target cac 20
> 
> You trace a careful sigil in the air, shaping the pattern of the Carrion Call spell.
> You begin to weave mana lines into a target pattern around a warklin mauler.
> 
> > 
> [combat-trainer]>app mauler quick
> 
> You are certain that the warklin mauler is battered.
> The warklin mauler has a relatively minor wound.
> You are certain that it is about as strong as you are.
> You are certain that it is about as agile as you are.
> You wonder if it is far less disciplined than you are.
> You are certain that it is rather less quick to react than you are.
> You are confident that it is not quite as conditioned as you are.
> Taking stock of its offensive abilities, and defending with a diamond-hide shield edged with a tyrium band and an etched steel parry stick with black leather straps, you are certain that the warklin mauler is definitely less skilled.
> Taking stock of its defensive abilities, and attacking with your fists, you are certain that the warklin mauler is a worthy opponent.
> 
> [Roundtime: 3 seconds]
> R> 
> [combat-trainer]>stalk
> 
> You're already stalking a warklin mauler!

intangibles commented:
Added @force_unhide setting

Also added in a quick combat fix so that if dance_action: is blank, it stops forcing line feeds.
fput(game_state.next_dance_action) if !game_state.next_dance_action.nil?

intangibles commented:
to do: add stop stalk to end of combat clean up

intangibles commented:
@rcuhljr 
@rpherbig 

I've been using this for quite awhile and everything seems to be working smoothly.  It's probably ready for a merge.

Also, I'm not really sure what the conflict is that popped up. with base-empty.yaml.

rpherbig commented on profiles/base-empty.yaml:
```diff
@@ -45,4 +45,5 @@ empty_values:
-  whirlwind_trainables: []
\ No newline at end of file
+  whirlwind_trainables: []
+  hide_type: hide
```
This should not be added here

rpherbig replied:
That's what caused the merge conflict

intangibles replied:
Hmm...

@rcuhljr suggested i add hide_type: hide  to the base-empty.   But he wasn't sure if maybe it was a better idea to just let it error out.

Or are you talking about the newline errors that I keep getting? Quite annoying!

intangibles replied:
I found some sublime settings to fix those trailing white spaces and endlines! hopefully gone forever.

intangibles commented:
Updated wiki.

hide_type: Value of "stalk" changes hide type to "slip stalk". This will perform the command "stalk" instead of "hide" which activates "slip stalk". "Slip stalk" is only available for a number of professions and must be learned. Default value: hide. Uses hide as normal.

force_unhide: Value of true will always unhide after hiding via Stealth timer under training_abilities:.

dont_stalk: Value of true prevents stalk usage after hiding when hiding via Stealth timer under training_abilities:. Default value is false.
